packages/starlight/user-components/LinkButton.astro

```diff
@@ -0,0 +1,72 @@
+---
+import type { HTMLAttributes } from 'astro/types';
+import { Icons } from '../components/Icons';
+import Icon from './Icon.astro';
+
+interface Props extends Omit<HTMLAttributes<'a'>, 'href'> {
+	href: string | URL;
+	icon?: keyof typeof Icons | undefined;
+	iconPlacement?: 'start' | 'end' | undefined;
+	variant?: 'primary' | 'secondary' | 'minimal';
+}
+
+const {
+	class: className,
+	icon,
+	iconPlacement = 'end',
+	variant = 'primary',
+	...attrs
+} = Astro.props;
+---
+
+<a class:list={['sl-link-button not-content', variant, className]} {...attrs}>
+	{icon && iconPlacement === 'start' && <Icon name={icon} size="1.5rem" />}
+	<slot />
+	{icon && iconPlacement === 'end' && <Icon name={icon} size="1.5rem" />}
+</a>
+
+<style>
+	.sl-link-button {
+		align-items: center;
+		border-radius: 999rem;
+		color: var(--sl-color-white);
+		display: inline-flex;
+		font-size: var(--sl-text-sm);
+		gap: 0.5em;
+		line-height: 1.7145;
+		padding: 0.5rem 1.125rem;
+		text-decoration: none;
+	}
+
+	.sl-link-button.primary {
+		background: var(--sl-color-text-accent);
+		border: 1px solid var(--sl-color-text-accent);
+		color: var(--sl-color-black);
+	}
+	.sl-link-button.secondary {
+		border: 1px solid;
+	}
+	.sl-link-button.minimal {
+		outline-offset: 0.25rem;
```

Comment:
Not strictly related to this PR, but noting it here while paying attention to this component: our default colours make focus styling quite hard to see. For example, one of these is focused but the difference is almost imperceptible:

<img width="340" alt="Two blue link buttons. The left button has a focus outline that is so close in colour to the button background that it is almost impossible to see." src="https://github.com/user-attachments/assets/12a73e8a-0bcc-412a-99e5-3a1446bad0aa">

The situation is slightly better in dark mode, but not much. I wonder if we could do something with `outline-offset` also for the other variants to help avoid people (and us) hitting this?

Reply:
There was already a `outline-offset: 0.25rem;` introduced in this PR for the minimal variant to avoid having the outline being on top of the text. We can just move this to the base `.sl-link-button` selector and this would fix it for all buttons.